### DISCUSSION TITLE
Undocumented faces

### DISCUSSION THREAD

**Discussion by BrianBinFL** [joeycastillo/Sensor-Watch#277] [Q&A] [answered]
I've looked through the [Watch Face Documentation](https://joeycastillo.github.io/Sensor-Watch-Documentation/watchfaces/) and it seems very well done. In addition to the faces documented on that page there are some dedicated pages for things like [Finetune and Nanosec](https://www.sensorwatch.net/docs/watchfaces/nanosec/) - also well done and very helpful.

But looking at the [Sensor Watch Builder page](https://sensor-watch-builder.fly.dev/) there are around 40 faces that I don't see see any documentation for (e.g. tachymeter_face). I've tried looking at the source code for some of these faces and for the most part the source code is not commented in a way that would tell someone how to use them, what they are for, etc.

Is there another repository of documentation somewhere for the faces not documented on the Watch Face Documentation page?

**Comment by wryun** [comment, marked as answer]
The best source of information is often the original PR that integrated them. e.g. for tachymeter:

https://github.com/joeycastillo/Sensor-Watch/pull/165

I'm sure if you made a PR to https://github.com/joeycastillo/sensorwatch.net it would be appreciated.

**Reply by BrianBinFL** [reply]
I figured out that "PR" means "pull request". Thanks for the link to the PR for Tachymeter. I'll look through the PRs for docs for the undocumented faces and collect them into one place for easy reference.

You said that if I made a PR to https://github.com/joeycastillo/sensorwatch.net it would be appreciated. I didn't understand that part. What would my PR be? You just mean if I developed a new face/feature for the watch? I do hope to do that someday.

Again, thanks for the pointer. I appreciate it!

**Reply by wryun** [reply]
Sorry, I should have been clearer! Sometimes there's just too much jargon.

That other repository is the documentation repository. i.e. if you can figure out how that works, you could edit the documentation to add the tachymeter documentation to sensorwatch.net

At least, speaking for myself, for the one and a bit faces I've developed, I've been too lazy to add them to the documentation yet. I can understand this is quite frustrating for users, but I'm happy that Joey has been generally accepting of faces going into the main repository even if they're not quite polished.

**Reply by BrianBinFL** [reply]
Thanks for the explanation Wryun. I would definitely like to help out with editing the documentation. I'll see if I can figure out how to do so.
